docs/usage/usage-guide.md

```diff
@@ -700,3 +700,208 @@ interface ThunkAPI {
+      usersAdapter.upsertMany(state, action.payload)
+    })
+  }
+})
+
+const reducer = slice.reducer
+export default reducer
+
+export const { removeUser } = slice.actions
+```
+
+#### Using selectors with `createEntityAdapter`
+
+The entity adapter providers a selector factory that generates the most common selectors for you. Taking the example above, we can add selectors to our `usersSlice` like this:
+
+```js
+const {
+  selectIds,
+  selectEntities,
+  selectAll,
+  selectTotal
+} = usersAdapter.getSelectors(state => state.users)
+
+export const selectUserIds = state => selectIds(state)
+export const selectUserEntities = state => selectEntities(state)
+export const selectAllUsers = state => selectAll(state)
+export const selectTotalUsers = state => selectTotal(state)
+```
+
+You could then use these selectors in a component like this:
+
+```js
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { selectTotalUsers, selectAllUsers } from './usersSlice'
+
+import styles from './UsersList.module.css'
+
+export function UsersList() {
+  const count = useSelector(selectTotalUsers)
+  const users = useSelector(selectAllUsers)
+
+  return (
+    <div>
+      <div className={styles.row}>
+        There are <span className={styles.value}>{count}</span> users.{' '}
+        {count === 0 && `Why don't you fetch some more?`}
+      </div>
+      {users.map(user => (
+        <div key={user.id}>
+          <div>{`${user.first_name} ${user.last_name}`}</div>
+          <div></div>
+        </div>
+      ))}
+    </div>
+  )
+}
+```
+
+To see this all working together, you can view the full code of this example usage on [CodeSandbox](https://codesandbox.io/s/rtk-entities-basic-example-4jg0m)
+
+#### Working with entities without an id property
+
+If your data set does not have an `id` property, createEntityAdapter offers a `selectId` argument that you can use.
+
+```js
+// In this instance, our user data always has a primary key of `idx`
+const userData = {
+  users: [
+    { idx: 1, first_name: 'Test' },
+    { idx: 2, first_name: 'Two' }
+  ]
+}
+
+// Being that our primary key is `idx` and not `id`, let the entity adapter know that with `selectId`
+export const usersAdapter = createEntityAdapter({
+  selectId: user => user.idx
+})
+```
+
+#### Sorting your entities by a default key
+
+`createEntityAdapter` provides a `sortComparer` argument that you can leverage to sort the collection of `ids` in state. This can be very useful for when you want to guarantee a sort order and your data doesn't come presorted.
+
+```js
+// In this instance, our user data always has a primary key of `idx`
+const userData = {
+  users: [
+    { id: 1, first_name: 'Test' },
+    { id: 2, first_name: 'Banana' }
+  ]
+}
+
+// Sort by `first_name`. `ids` would be ordered like `ids: [ 2, 1 ]` being that B comes before T
+export const usersAdapter = createEntityAdapter({
+  sortComparer: (a, b) => a.first_name.localeCompare(b.first_name)
+})
+
```

Comment:
Add a comment here that since it's specifically `state.ids` that's being sorted, you would do `state.ids.map()` to get the items in sorted order, and `selectAll` does that specifically.